code/ui/deadhud/SpectatedPlayerInfo.cs

```diff
@@ -0,0 +1,130 @@
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+using TTTReborn.Player;
+using TTTReborn.Player.Camera;
+using TTTReborn.Roles;
+
+namespace TTTReborn.UI
+{
+    public class SpectatedPlayerInfo : Panel
+    {
+        private TTTPlayer _currentPlayer;
+        private NamePanel _namePanel;
+        private IndicatorsPanel _indicatorsPanel;
+
+        public SpectatedPlayerInfo()
+        {
+            StyleSheet.Load("/ui/deadhud/SpectatedPlayerInfo.scss");
+
+            _namePanel = new NamePanel(this);
+            _indicatorsPanel = new IndicatorsPanel(this);
+        }
+
+        public override void Tick()
+        {
+            base.Tick();
+
+            if (Local.Pawn is not TTTPlayer player || player.Camera is not SpectateCamera spectateCamera)
+            {
+                SetClass("hide", true);
+
+                return;
+            }
+
+            SetClass("hide", spectateCamera.TargetPlayer == null);
+
+            if (_currentPlayer != spectateCamera.TargetPlayer)
+            {
+                _currentPlayer = spectateCamera.TargetPlayer;
+
+                _namePanel.SetTargetPlayer(_currentPlayer);
+                _indicatorsPanel.SetTargetPlayer(_currentPlayer);
+            }
+        }
+
+        private class NamePanel : Panel
+        {
+            private TTTPlayer _currentPlayer;
+            private readonly Label _nameLabel;
+
+            public NamePanel(Panel parent)
+            {
+                Parent = parent;
+
+                _nameLabel = Add.Label("Unknown player", "namelabel");
+            }
+
+            public void SetTargetPlayer(TTTPlayer player)
+            {
+                _currentPlayer = player;
+            }
+
+            public override void Tick()
+            {
+                base.Tick();
+
+                if (_currentPlayer == null)
+                {
+                    return;
+                }
+
+                if (_currentPlayer.Role is NoneRole)
+                {
+                    Style.BackgroundColor = Color.Black;
+                }
+                else
+                {
+                    Style.BackgroundColor = _currentPlayer.Role.Color;
+                }
+
+                Style.Dirty();
+
+                _nameLabel.Text = $"{_currentPlayer.GetClientOwner()?.Name}";
+            }
+        }
+
+        private class IndicatorsPanel : Panel
```

Comment:
This is a copy and paste of how we do the health HUD on an alive player right? Is there any way to just reuse the code?

Reply:
We could put it into a shared folder, but that's quite ugly. Additionally, we don't know what we gonna change in the future. One thing we could do is to use the same scss.